packages/expo-asset/src/Asset.js

```diff
@@ -167,8 +202,12 @@ export default class Asset {
   }
 
   static fromModule(moduleId) {
```

Comment:
I think it'd be better to keep this method focused on loading from module IDs. Can the caller be changed to call fromURI / fromModule depending on what it has?

Reply:
@ide sounds reasonable, the reason I did this is because if you were to run the same code on web you would have different results:
```js
const module = require('./image.png');
// On Native you'd get some number like 2
// On Web you'd get base64
```
So a common operation would also have to do a platform check each time.

```js
const asset = Asset.fromModule(require('...'));
await asset.downloadAsync();

// Becomes

let asset;
if (Platform.OS === "web") {
  asset = Asset.fromURI(require('...'));
} else {
  asset = Asset.fromModule(require('...'));
}
await asset.downloadAsync();
```

Reply:
I see, `Asset.fromModule(require(X))` definitely should work here. TBH it's not great IMO that you get a base64-encoded string on web either, which isn't cacheable and is less efficiently compressed, so I think we want to get away from that eventually. That said, IIRC Webpack plugins that handle image-requires make them return URLs so I think what you have here is right. Might be good to rename the parameter:

```suggestion
  static fromModule(virtualAssetModule) {
    // The virtual asset module may be an opaque ID (Metro) or a (Base64) URI (Webpack)
```